File: 204 Count Primes/solution.py
class Solution:
    # @param {integer} n
    # @return {integer}
    def countPrimes(self, n):
        result = [True]*max(2,n)
        result[0] = False
        result[1] = False
        init = 2
        while init*init<n:
            if result[init]:
                mark = init*init
                while mark<n:
                    result[mark] = False
                    mark+=init
            init+=1
        return sum(result)
        

File: 204 Count Primes/solution.cpp
class Solution {
public:
    int countPrimes(int n) {
        bool isprime[n];
        for (int i=0;i<n;i++){
            isprime[i] = true;
        }
        isprime[0] = isprime[1] = false;
        int start = 2;
        int count = 0;
        while (start*start<n){
            if (isprime[start]){
                int mark = start*start;
                while (mark<n){
                    isprime[mark] = false;
                    mark+=start;
                }
            }
            start+=1;
        }
        for (int i=0;i<n;i++){
            if (isprime[i])
                count++;
        }
        return count;
    }
};